tests/test_updater_validation.py

```diff
@@ -0,0 +1,107 @@
+"""
+
+import os
+import sys
+import tempfile
+import unittest
+
+import securesystemslib.exceptions
+
+from tests import utils
+from tests.repository_simulator import RepositorySimulator
+from tuf import unittest_toolbox
+from tuf.api.metadata import TargetFile
+from tuf.exceptions import FetcherHTTPError
+from tuf.ngclient import Updater
+
+
+class TestUpdater(unittest_toolbox.Modified_TestCase):
+    """Test ngclient Updater input validation."""
+
+    def setUp(self) -> None:
+        # pylint: disable-next=consider-using-with
+        self.temp_dir = tempfile.TemporaryDirectory()
+        self.metadata_dir = os.path.join(self.temp_dir.name, "metadata")
+        self.targets_dir = os.path.join(self.temp_dir.name, "targets")
+        os.mkdir(self.metadata_dir)
+        os.mkdir(self.targets_dir)
+
+        # Setup the repository, bootstrap client root.json
+        self.sim = RepositorySimulator()
+        with open(os.path.join(self.metadata_dir, "root.json"), "bw") as f:
+            f.write(self.sim.signed_roots[0])
+
+    def tearDown(self) -> None:
+        self.temp_dir.cleanup()
+
+    def _new_updater_with_targets_setup(self) -> Updater:
+        return Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            self.targets_dir,
+            "https://example.com/targets/",
+            fetcher=self.sim,
+        )
+
+    def _new_updater_without_targets_setup(self) -> Updater:
+        return Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            fetcher=self.sim,
+        )
+
+    def _new_updater_with_non_existing_metadata_dir(self) -> Updater:
+        return Updater(
+            "non_existing_metadata_dir",
+            "https://example.com/metadata/",
+            fetcher=self.sim,
+        )
+
+    def test_empty_target_path_validation(self) -> None:
+        updater = self._new_updater_with_targets_setup()
+        target_file = updater.get_targetinfo("")
+        self.assertIsNone(target_file)
+
+    def test_empty_targetinfo_validation(self) -> None:
+        updater = self._new_updater_without_targets_setup()
+        info = TargetFile(1, {"sha256": ""}, "")
+        with self.assertRaisesRegex(
+            ValueError, "target_dir must be set if filepath is not given"
+        ):
+            updater.download_target(info)
+
+        updater = self._new_updater_with_targets_setup()
+        with self.assertRaisesRegex(FetcherHTTPError, "No target"):
+            updater.download_target(info)
+
+    def test_empty_filepath_validation(self) -> None:
+        self.sim.add_target("targets", b"content", "targetpath")
+        self.sim.targets.version += 1
+        self.sim.update_snapshot()
+
+        updater = self._new_updater_with_targets_setup()
+        target_info = updater.get_targetinfo("targetpath")
+        with self.assertRaises(securesystemslib.exceptions.StorageError):
+            if target_info is not None:
```

Comment:
Why do you need to make that check?
When `target_info` is `None` considering you are adding it at the begging of the test?

Reply:
@MVrachev this one is just for the linter 😄 It fails for non-matching object types - `Optional` vs `TargetFile`. On the other side, I found it as a good correction, as `download_target` doesn't expect None values.

Reply:
nit: An alternative that I saw in another test is

```python
        updater = self._new_updater_with_targets_setup()
        target_info = updater.get_targetinfo("targetpath")
        assert target_info is not None
        with self.assertRaises(securesystemslib.exceptions.StorageError):
             updater.download_target(target_info, filepath="")
```

Reply:
I was going to link to it.
I think it looks better as we already use asserts in our tests.